proxy.go

```diff
@@ -243,8 +248,17 @@ func unsizedKind(v reflect.Value) reflect.Kind {
 }
 
 func valueOfProxy(L *lua.State, idx int) (reflect.Value, reflect.Type) {
-	proxy := (*valueProxy)(L.ToUserdata(idx))
-	return proxy.v, proxy.t
+	proxyId := *(*uintptr)(L.ToUserdata(idx))
+
+	proxymu.RLock()
+	val, prs := proxyMap[proxyId]
```

Comment:
Just a nit for consistency with the rest of the code (and as used by the official Go doc): rename `prs` to `ok`.